tests/unit/test_upstream_dataset.py

```diff
@@ -0,0 +1,110 @@
+        "name": "Project_1_unified_dataset_dedup_features",
+        "description": "Features for all the rows and values in the source dataset. Used in Dedup Workflow.",
+        "version": "543",
+        "keyAttributeNames": ["entityId"],
+        "tags": [],
+        "created": {
+            "username": "admin",
+            "time": "2019-06-05T18:31:59.327Z",
+            "version": "212",
+        },
+        "lastModified": {
+            "username": "admin",
+            "time": "2019-07-18T14:19:28.133Z",
+            "version": "22225",
+        },
+        "relativeId": "datasets/12",
+        "upstreamDatasetIds": ["unify://unified-data/v1/datasets/8"],
+        "externalId": "Project_1_unified_dataset_dedup_features",
+    }
+
+    dataset_json = {
+        "id": "unify://unified-data/v1/datasets/12",
+        "name": "Project_1_unified_dataset_dedup_features",
+        "description": "Features for all the rows and values in the source dataset. Used in Dedup Workflow.",
+        "version": "543",
+        "keyAttributeNames": ["entityId"],
+        "tags": [],
+        "created": {
+            "username": "admin",
+            "time": "2019-06-05T18:31:59.327Z",
+            "version": "212",
+        },
+        "lastModified": {
+            "username": "admin",
+            "time": "2019-07-18T14:19:28.133Z",
+            "version": "22225",
+        },
+        "relativeId": "datasets/12",
+        "upstreamDatasetIds": ["unify://unified-data/v1/datasets/8"],
+        "externalId": "Project_1_unified_dataset_dedup_features",
+    }
+
+    upstream_json = ["unify://unified-data/v1/datasets/8"]
+
+    upstream_ds_json = {
+        "id": "unify://unified-data/v1/datasets/8",
+        "name": "Project_1_unified_dataset",
+        "description": "",
+        "version": "529",
+        "keyAttributeNames": ["tamr_id"],
+        "tags": [],
+        "created": {
+            "username": "admin",
+            "time": "2019-06-05T16:28:11.639Z",
+            "version": "83",
+        },
+        "lastModified": {
+            "username": "admin",
+            "time": "2019-07-22T20:31:23.968Z",
+            "version": "23146",
+        },
+        "relativeId": "datasets/8",
+        "upstreamDatasetIds": ["unify://unified-data/v1/datasets/6"],
+        "externalId": "Project_1_unified_dataset",
+    }
+
+    upstream_datasets = {
+        "relative_id": "datasets/8",
+        "name": "Project_1_unified_dataset",
+        "version": "529",
+    }
+
+    unify = Client(UsernamePasswordAuth("username", "password"))
+
+    url_prefix = "http://localhost:9100/api/versioned/v1/"
+    project_url = url_prefix + "projects/1"
+    datasets_url = url_prefix + "datasets"
+    dataset_url = url_prefix + "datasets/12"
+    upstream_url = url_prefix + "datasets/12/upstreamDatasets"
+    upstream_ds_url = url_prefix + "datasets/8"
+
+    responses.add(responses.GET, project_url, json=project_config)
```

Comment:
I don't think you make this request or the one to datasets_url (also, the response body for datasets_url wouldn't make sense as it would need to be a list of datasets)